core/variant/method_ptrcall.h

```diff
@@ -94,12 +99,13 @@ struct PtrToArg {};
 		_FORCE_INLINE_ static m_type convert(const void *p_ptr) {             \
 			return *reinterpret_cast<const m_type *>(p_ptr);                  \
 		}                                                                     \
+		typedef m_type EncodeT;                                               \
 		_FORCE_INLINE_ static void encode(const m_type &p_val, void *p_ptr) { \
 			*((m_type *)p_ptr) = p_val;                                       \
 		}                                                                     \
 	}
 
-MAKE_PTRARG(bool);
+MAKE_PTRARGCONV(bool, uint32_t);
 // Integer types.
 MAKE_PTRARGCONV(uint8_t, int64_t);
 MAKE_PTRARGCONV(int8_t, int64_t);
```

Comment:
Since the new extension API is a breaking change, can we take the opportunity to change this? `GetTypeInfo` has included info about int size and sign for quite some time now, so this conversion should not be needed anymore if the bindings use that info (as already do the C# bindings).

Reply:
I added this information in the bindings already, but this is mostly to create the right ones on the exposed API. The ptrcall API will still send everything through the biggest possible format (int64)

Reply:
Also, the GDScript interpreter will use ptrcall if it knows the types now, and it always sends int64_t, its not possible to call the "right" one directly.